pkgs/applications/graphics/apitrace/default.nix

```diff
@@ -15,7 +15,42 @@ stdenv.mkDerivation rec {
   # of games -- so it's fine to use e.g. bundled snappy.
   buildInputs = [ libX11 procps python2 libdwarf qtbase qtwebkit ];
 
-  nativeBuildInputs = [ cmake ];
+  nativeBuildInputs = [ cmake wrapQtAppsHook ];
+
+  # Don't automatically wrap all binaries, I prefer to explicitly only wrap
+  # `qapitrace`.
+  dontWrapQtApps = true;
+
+  postFixup = ''
+    # Since https://github.com/NixOS/nixpkgs/pull/60985 OpenGL programs on NixOS are
+    # finding the opengl drivers in /run/opengl-driver thank to patched
+    # runpath, using `addOpenGLRunpath`. If they use a library loader such as
+    # `libglvnd`, this logic is already included in the library loader.
+    #
+    # Apitrace wraps programs using LD_PRELOAD wrappers (in $out/lib/apitrace/wrappers).
+    # Theses wrappers replace the library loader used by the original program
+    # by overriding `dlopen`.
+    #
+    # In order to find the correct OpenGL library, apitrace wrappers should
+    # have the correct rpath. This is done in the following lines.
+    # To stay simple, we add paths for 32 and 64 bits unconditionally. This
+    # have no impact on the final closure size.
+    for i in $out/lib/apitrace/wrappers/*.so
+    do
+      echo "Patching OpenGL driver path for $i"
+      patchelf --set-rpath "/run/opengl-driver/lib:/run/opengl-driver-32/lib:$(patchelf --print-rpath $i)" $i
+    done
+
+    # Theses open the OpenGL driver at runtime, but it is not listed as NEEDED libraries. They need
+    # a reference to libglvnd and libGL.
+    for i in $out/bin/eglretrace $out/bin/glretrace
+    do
+      echo "Patching RPath for $i"
+      patchelf --set-rpath "${stdenv.lib.makeLibraryPath [libglvnd libGL]}:$(patchelf --print-rpath $i)" $i
```

Comment:
We might only need `libglvnd` here - `${libGL}/lib` is just a symlink to `${libglvnd}/lib`.

Reply:
Done.